app/controllers/declaration/DestinationCountriesController.scala

```diff
@@ -50,139 +47,89 @@ class DestinationCountriesController @Inject()(
     extends FrontendController(mcc) with I18nSupport with ModelCacheable {
 
   def displayPage(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
-    request.cacheModel.locations.destinationCountries match {
-      case Some(data) =>
-        request.declarationType match {
-          case DeclarationType.SUPPLEMENTARY => Redirect(routes.DestinationCountryController.displayPage())
-          case DeclarationType.STANDARD | DeclarationType.SIMPLIFIED =>
-            Ok(
-              destinationCountriesStandardPage(
-                mode,
-                Standard.form(data.countryOfDispatch, data.countryOfDestination).fill(data),
-                data.countriesOfRouting
-              )
-            )
-        }
-      case None => Redirect(controllers.declaration.routes.OriginationCountryController.displayPage(mode))
+    request.declarationType match {
+      case DeclarationType.SUPPLEMENTARY => Redirect(routes.DestinationCountryController.displayPage())
```

Comment:
Is this an infinite loop?

Reply:
We probably need a `ReturnToStartException` for this kind of thing

Reply:
`DestinationCountryController` is different controller. I know stupid naming, but this Controller will disappear in next PR, so I decided to don't refactor this. Whole this controller now support just one input `countriesOfRouting` and will be changed and renamed